gradio/pipelines.py

```diff
@@ -18,253 +18,579 @@ def load_from_pipeline(pipeline: pipelines.base.Pipeline) -> dict:
-                components.Label(label="Score", render=False),
-            ],
-            "preprocess": lambda c, q: {"context": c, "question": q},
-            "postprocess": lambda r: (r["answer"], r["score"]),
-        }
-    elif hasattr(transformers, "SummarizationPipeline") and isinstance(
-        pipeline, pipelines.text2text_generation.SummarizationPipeline
-    ):
-        pipeline_info = {
-            "inputs": components.Textbox(lines=7, label="Input", render=False),
-            "outputs": components.Textbox(label="Summary", render=False),
-            "preprocess": lambda x: {"inputs": x},
-            "postprocess": lambda r: r[0]["summary_text"],
-        }
-    elif hasattr(transformers, "TextClassificationPipeline") and isinstance(
-        pipeline, pipelines.text_classification.TextClassificationPipeline
-    ):
-        pipeline_info = {
-            "inputs": components.Textbox(label="Input", render=False),
-            "outputs": components.Label(label="Classification", render=False),
-            "preprocess": lambda x: [x],
-            "postprocess": lambda r: {i["label"].split(", ")[0]: i["score"] for i in r},
-        }
-    elif hasattr(transformers, "TextGenerationPipeline") and isinstance(
-        pipeline, pipelines.text_generation.TextGenerationPipeline
-    ):
-        pipeline_info = {
-            "inputs": components.Textbox(label="Input", render=False),
-            "outputs": components.Textbox(label="Output", render=False),
-            "preprocess": lambda x: {"text_inputs": x},
-            "postprocess": lambda r: r[0]["generated_text"],
-        }
-    elif hasattr(transformers, "TranslationPipeline") and isinstance(
-        pipeline, pipelines.text2text_generation.TranslationPipeline
-    ):
-        pipeline_info = {
-            "inputs": components.Textbox(label="Input", render=False),
-            "outputs": components.Textbox(label="Translation", render=False),
-            "preprocess": lambda x: [x],
-            "postprocess": lambda r: r[0]["translation_text"],
-        }
-    elif hasattr(transformers, "Text2TextGenerationPipeline") and isinstance(
-        pipeline, pipelines.text2text_generation.Text2TextGenerationPipeline
-    ):
-        pipeline_info = {
-            "inputs": components.Textbox(label="Input", render=False),
-            "outputs": components.Textbox(label="Generated Text", render=False),
-            "preprocess": lambda x: [x],
-            "postprocess": lambda r: r[0]["generated_text"],
-        }
-    elif hasattr(transformers, "ZeroShotClassificationPipeline") and isinstance(
-        pipeline, pipelines.zero_shot_classification.ZeroShotClassificationPipeline
-    ):
-        pipeline_info = {
-            "inputs": [
-                components.Textbox(label="Input", render=False),
-                components.Textbox(
-                    label="Possible class names (" "comma-separated)", render=False
+    if "transformers.pipelines" in str(type(pipeline)):
```

Comment:
Maybe refactor these top-level `if` branches into separate shorter helper functions like `_handle_transformers_pipeline()`?

Also, what if `str(type(pipeline))` is `my_happy_transformers.pipelines.something`? This should probably be `.startswith()`.